pandas/tests/reshape/test_pivot.py

```diff
@@ -64,6 +64,25 @@ def test_pivot_table(self):
             index + [columns])['D'].agg(np.mean).unstack()
         tm.assert_frame_equal(table, expected)
 
+    def test_pivot_table_categorical_observed_equal(self, observed):
+        # issue #24923
+        df = pd.DataFrame({'col1': list('abcde'),
+                           'col2': list('fghij'),
+                           'col3': [1, 2, 3, 4, 5]})
+
+        df.col1 = df.col1.astype('category')
+        df.col2 = df.col1.astype('category')
+
+        expected = df.pivot_table(index='col1', values='col3',
+                                  columns='col2', aggfunc=np.sum,
+                                  fill_value=0)
```

Comment:
this example needs to be different; compute the expected w/o using the categoricals, then set the categoricals on the output.

This should match of pivoting with the categoricals as you have now. IOW the result you have is fine, but the expected needs to change.

Reply:
Have done this - expected is pivoted without categorical datatype and set afterwards, whereas result is pivoted based on categorical datatype input and they both are equivalent.